openhands/server/session/agent.py

```diff
@@ -63,9 +65,27 @@ async def start(
             raise RuntimeError(
                 'Session already started. You need to close this session and start a new one.'
             )
-        await self._create_security_analyzer(config.security.security_analyzer)
-        await self._create_runtime(runtime_name, config, agent)
-        await self._create_controller(
+        
+        # Runtimes can take a long time to create / start - and we don't want to lock up the 
+        # main event loop waiting for this, so we do this in a background thread
+        # We may consider making the executor shared and global if it is used very frequently.
+        with ThreadPoolExecutor(1) as executor: 
+            await asyncio.get_event_loop().run_in_executor(executor, self._start, runtime_name, config, agent, max_iterations, max_budget_per_task, agent_to_llm_config, agent_configs)
+        if self.controller is not None:
+            self.controller.agent_task = asyncio.create_task(self.controller.start_step_loop())
```

Comment:
Should the step loop run in its own thread?

Right now the HTTP endpoints seem to hang while the agent is doing things...

Reply:
Ooh - you're right - I didn't catch that - looks like I need to update this PR. (Tested with a "wait 5 seconds" and saw that the test script had 10 seconds between requests)